packages/main/src/components/OverflowToolbar/index.tsx

```diff
@@ -0,0 +1,207 @@
+      children: newChildren,
+      popoverElements,
+      previousWidth
+    });
+    if (Children.count(this.props.children) !== Children.count(this.state.children)) {
+      if (
+        this.toolbarRef.current.clientWidth === this.toolbarRef.current.scrollWidth &&
+        this.toolbarRef.current.clientWidth >= this.state.previousWidth[0]
+      ) {
+        this.addToolbarItem(newChildren, popoverElements, previousWidth, renderToggle);
+      }
+    } else {
+      this.setState({ renderToggle: false });
+    }
+  }
+
+  render() {
+    const { width, align, toolbarDesign, classes, className, style, tooltip } = this.props;
+    const rootClasses = StyleClassHelper.of(classes.toolbarRoot);
+    const overflowClasses = StyleClassHelper.of(classes.overflowRoot);
+
+    switch (align) {
+      case ToolbarAlignment.Start:
+        rootClasses.put(classes.toolbarAlignStart);
+        break;
+      case ToolbarAlignment.End:
+        rootClasses.put(classes.toolbarAlignEnd);
+        break;
+      case ToolbarAlignment.Middle:
+        rootClasses.put(classes.toolbarAlignMiddle);
+        break;
+      case ToolbarAlignment.SpaceBetween:
+        rootClasses.put(classes.toolbarAlignSpaceBetween);
+        break;
+      default:
+        rootClasses.put(classes.toolbarAlignStart);
+    }
+
+    switch (toolbarDesign) {
+      case ToolbarDesign.ContentBar:
+        rootClasses.put(classes.contentBar);
+        overflowClasses.put(classes.contentBar);
+        break;
+      case ToolbarDesign.PageFooter:
+        rootClasses.put(classes.pageFooter);
+        overflowClasses.put(classes.pageFooter);
+        break;
+      case ToolbarDesign.ContainerBar:
+        rootClasses.put(classes.containerBar);
+        overflowClasses.put(classes.containerBar);
+        break;
+      case ToolbarDesign.ContentBarTransparent:
+        rootClasses.put(classes.contentBarTransparent);
+        overflowClasses.put(classes.contentBarTransparent);
+        break;
+      default:
+        rootClasses.put(classes.containerBar);
+        overflowClasses.put(classes.containerBar);
+    }
+
+    if (className) {
+      rootClasses.put(className);
+    }
+
+    const inlineStyle = { width };
+    if (style) {
+      Object.assign(inlineStyle, style);
+    }
+    const PopoverWrapper = () => {
+      if (this.state.renderToggle) {
+        return (
+          <Popover
+            style={{ marginLeft: 'auto' }}
+            key="popover"
+            openBy={<Button icon="overflow" key={'overflowButton'} />}
+            noHeader={true}
+            placementType={PlacementType.Bottom}
+          >
+            <div style={{ display: 'flex', flexDirection: 'column' }}>{this.state.popoverElements}</div>
+          </Popover>
+        );
+      }
+      return null;
+    };
+    return (
+      <div className={rootClasses.valueOf()} style={{ ...inlineStyle }} title={tooltip} ref={this.toolbarRef}>
+        {this.state.children}
+        <PopoverWrapper />
```

Comment:
I would move this content of the PopoverWrapper function right into here by using `{this.state.renderToggle && <Popover>...}`